### PR TITLE
Week2 - HW

### PR DESCRIPTION
**Pull request by Saikiruthiga**

### CODE REVIEW

File: databases/week2/Week2-HW.sql [hunk 1]
```diff
@@ -0,0 +1,358 @@
...
+        'Studying',
+        'Complete the week3 database preparation',
+        '2024-07-08 00:23:58',
+        '2024-07-10 12:00:25',
+        '2025-07-08 00:00:23',
+        1,
+        1
+    );
+-- b) Change the title of a task
+update task set title = 'Learning' ORDER BY updated DESC limit 1;
```
Comment (NatsGt) [inline databases/week2/Week2-HW.sql]: Try to specify an id. If you do it this way, in a real application, another user might insert new data to the table and you will be updating a different task to the one you intended.

File: databases/week2/Week2-HW.sql [hunk 2]
```diff
@@ -0,0 +1,358 @@
...
+-- c) Change a task due date
+update task
+set
+    due_date = '2024-12-12 00:02:23'
+where
+    id = 36;
+-- d) Change a task status
+update task set status_id = 2 ORDER BY updated desc limit 1;
```
Comment (NatsGt) [inline databases/week2/Week2-HW.sql]: Same comment as above

File: databases/week2/Week2-HW.sql [hunk 3]
```diff
@@ -0,0 +1,358 @@
...
+-- e) Mark a task as complete
+update task set status_id = 3 ORDER BY updated desc limit 1;
```
Comment (NatsGt) [inline databases/week2/Week2-HW.sql]: Same comment as above

File: databases/week2/Week2-HW.sql [hunk 4]
```diff
@@ -0,0 +1,358 @@
...
+insert into
+    student_1
+values (
+        3,
+        'Smith',
+        'smith_01@gmail.com',
+        '136856985',
+        2
+    );
+-- Part 3: More queries
```
Comment (NatsGt) [inline databases/week2/Week2-HW.sql:242]: The queries from this part seem fine BUT you used other tables from the ones specified on the assignment.

**Review by NatsGt — CHANGES_REQUESTED** [review]
Very good job. I just have some comments on your queries, also you seemed to use different tables for the part 3, check the instructions. The ER diagram for the author/tag/article seems correct. But the ER diagram for the restaurant needs to improve, I think the table between menu and customers is not necessary.

**Comment by Saikiruthiga** [comment]
Thank you for your review. I have changed the queries based on your comments and changed the restuarant erd

File: databases/week2/Week2-HW.sql (hunk 1 above)
Comment (Saikiruthiga) [reply databases/week2/Week2-HW.sql]: Yes, Agreed with your points. selected the task based on id

File: databases/week2/Week2-HW.sql (hunk 2 above)
Comment (Saikiruthiga) [reply databases/week2/Week2-HW.sql]: updated

File: databases/week2/Week2-HW.sql (hunk 3 above)
Comment (Saikiruthiga) [reply databases/week2/Week2-HW.sql]: updated

File: databases/week2/Week2-HW.sql (hunk 4 above)
Comment (Saikiruthiga) [reply databases/week2/Week2-HW.sql:242]: Yes, Sorry for my overlook. I have updated the database (user_task table) and changed the queries based on that

**Review by NatsGt — APPROVED** [review]

**Pull request merged**